common/src/test/java/io/pravega/common/util/IdGeneratorTest.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.common.util;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+
+public class IdGeneratorTest {
+
+    @Test
+    public void getRequesterId() {
+        long id1 = IdGenerator.getRequesterId();
+        long id2 = IdGenerator.getRequesterId();
+        assertEquals("Requester Id should be incremented", id1 >> Integer.SIZE, (id2 >> Integer.SIZE) -1);
```

Comment:
Are you sure this is a thorough enough test?

Are there other boundary cases to test? Such as numeric overflow? Negative-to-positive transitions?

Reply:
updated the api and added tests for numeric overflows.